tike/ptycho.py

```diff
@@ -95,9 +95,9 @@
 
 
 def _ptycho_interface(data, data_min,
```

Comment:
I agree that this variable naming system is slightly ambiguous. However, this naming system is explained in the module level docstring. A clearer name would be `data_min_corner`, and I would be happy to adopt this naming system instead. However, you have often told me that short variable names are better, so that's why I chose `data_min` over `data_min_corner`.

Here is why I didn't choose any of your suggested alternatives:
- `axis_min`, `det_min` The corners of multiple things need to be specified, so these options are more confusing than the current option because they remove the connection implied by using the word `data` in both variable names.
- `data_corner` is probably equally ambiguous because then you need to ask "which corner?" There 4 options. However, `data_min` is shorter.

Reply:
I'd be more inclined towards ```data_corner``` and explaining that it is the minimum. My second best is ```data_min_corner```. 

As soon as all corners are consistent, it doesn't matter which corner is picked. So, maybe we find an automated or safe way for this. How about a correction with the minimum values of h and v of the whole stack?

Reply:
> As soon as all corners are consistent, it doesn't matter which corner is picked.

This is not true. The alignment algorithms in `pytcho.combine_grids` rely on knowing that the corner is the min corner. Also, there is no way of determining which corner has been specified from the coordinates of the corners themselves.

I oppose the ability choose your own corner. This would add unneeded complexity to the code base when preprocessing the data to conform to our min corner specification is trivial.

Reply:
I would advocate that the quote still is true. Assume a grid of size (40, 60) and when you pick corners as (0, 0), (0, 20), (20, 0) or (20, 20) for min is zero. If I used a different corner these points will be (40, 60), (40, 80), (60, 60) or (60, 80). Then you subtract min(h-axis)=40 and min(v-axis)=60 from each axis.

I agree with you that this will add some unnecessary complexity, but also it will reduce the complexity in naming. It's your choice, but ```corner``` is a good thing to have in the variable.

Reply:
Two corners are required to define a rectangle. If I only tell you the side lengths and the location of one corner, you cannot draw a box correctly. It's impossible for you to know which corner I have given you without more information.

I agree that having `corner` in the variable name is good. I'm only debating whether corner naming consistency is enough information to correctly arrange boxes.

Reply:
Wait a minute.. I'm confused. Why do we need this variable at all? Isn't ```h``` and ```v``` are defined on the global coordinates?

Reply:
We need `data_corner` if the probe movement is independent of the detector, which it could be, but right now that feature is not implemented.

Reply:
Detector's are always fixed. So it is always the sample (or the probe in relative to the object) is moving. So, we may never need this argument.

Reply:
Probe movement will be on the order of 10s of microns at most, and your detector size 10s of centimeters. I think it is still on the negligible range.

Reply:
Good to know! I've removed that parameter from all public functions, so approve this pull please.